### DISCUSSION TITLE
git.execute with "--quiet" flag causes hang

### DISCUSSION THREAD

**Discussion by batabek** [gitpython-developers/GitPython#1316] [Q&A] [answered]
Hello,

I am trying to execute a commit using 

`git.execute(["git","commit","--quiet"])`

to suppress the message but it seems like it hangs. How can I fix it ?

using GitPython v.3.1.18

**Comment by Byron** [comment, marked as answer]
In order to execute git in the correct repository you would probably do it [as described here](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly). If that doesn't resolve the problem it's possible to [debug the git invocations](https://gitpython.readthedocs.io/en/stable/tutorial.html#git-command-debugging-and-customization) as well.

**Reply by batabek** [reply]
@Byron Thanks for the response. I have omitted the init. code so I guarantee that it's repo is initialized properly. This hang might  be related to this  [issue](https://github.com/gitpython-developers/GitPython/issues/72) ?

**Reply by Byron** [reply]
Probably not.

Calling `git.execute()` really isn't recommended and I wonder if following the docs leads to more idiomatic and easier to debug code.
